clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1776,10 +1777,13 @@ public void close() {
     }
 
     /**
-     * Tries to close the consumer cleanly within the specified timeout. This method waits up to
-     * {@code timeout} for the consumer to complete pending commits and leave the group.
+     * This method has been deprecated since Kafka 4.0 and should use {@link KafkaConsumer#close(CloseOptions)} instead.
+     * <p>
+     * Close the consumer with {@link CloseOptions.GroupMembershipOperation#DEFAULT default membership operation}
+     * cleanly within the specified timeout. This method waits up to
+     * {@code timeout} for the consumer to complete pending commits and may leave the group (if applicable).
      * If auto-commit is enabled, this will commit the current offsets if possible within the
-     * timeout. If the consumer is unable to complete offset commits and gracefully leave the group
+     * timeout. If the consumer is unable to complete offset commits and gracefully leave the group (if applicable)
```

Comment:
(if applicable) is ok to use here as long as we specified when it's applicable earlier in the javadocs